efibootmgr/reseal.go

```diff
@@ -340,6 +340,7 @@ func TrustCurrentBoot(assets *TrustedAssets, esp string) error {
 		data, ok := event.Data.(*tcglog.EFIImageLoadEvent)
 		if !ok {
 			log.Println("Invalid event data for EV_EFI_BOOT_SERVICES_APPLICATION event")
+			continue
 		}
```

Comment:
I've just thought actually - this catches the malformed kernel event measured by shim, which is malformed because the device path is completely empty.

It's entirely possible though for firmware to create a valid device path consisting of just a single termination node, and these nodes are omitted by go-efilib's efi.DevicePath. This means that `len(data.DevicePath)` could legitimately be zero and the subsequent code would crash.